lms/views/lti/basic_launch.py

```diff
@@ -294,8 +282,48 @@ def _show_document(
             extra=assignment_extra,
         )
 
+        # An assignment has been configured in the LMS as "gradable" if it has
+        # the `lis_outcome_service_url` param
+        assignment_gradable = bool(
+            self.context.lti_params.get("lis_outcome_service_url")
+        )
+
+        # Set up the JS config for the front-end
+        self._configure_js_to_show_document(document_url, assignment_gradable)
+
         return {}
 
+    def _configure_js_to_show_document(self, document_url, assignment_gradable):
+        # Only show the grading interface to teachers who aren't in Canvas, as
+        # Canvas uses its own built in Speedgrader
+        if (
+            assignment_gradable
+            and self.request.lti_user.is_instructor
+            and not self.context.is_canvas
+        ):
+            self.context.js_config.enable_grading()
+
+        # For students in Canvas with grades to submit we need to enable
+        # Speedgrader settings for gradable assignments
+        if (
+            assignment_gradable
+            and self.context.is_canvas
+            # `lis_result_sourcedid` associates a specific user with an
+            # assignment. This is evidence that a student is launching us
+            and self.context.lti_params.get("lis_result_sourcedid")
+        ):
+            self.context.js_config.add_canvas_speedgrader_settings(document_url)
```

Comment:
It looks like there's some way of usefully combo-ing the if statements here, but I don't think there actually is. I _think_ it's best to just separate these out and have them both state what they need.

Reply:
Yeah, it's tempting to shuffle these around trying to find conditional enlightenment but it's what's it is.

Nice that this happens here instead of in various `maybe_` methods far away.